exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScanStatistics.java

```diff
@@ -115,7 +118,7 @@ public void collect(Collection<T> metadataList) {
           previousCount.setValue(Statistic.NO_COLUMN_STATS);
         }
         ColumnMetadata columnMetadata = SchemaPathUtils.getColumnMetadata(schemaPath, metadata.getSchema());
-        TypeProtos.MajorType majorType = columnMetadata != null ? columnMetadata.majorType() : null;
+        TypeProtos.MajorType majorType = columnMetadata != null ? columnMetadata.majorType() : NULL;
```

Comment:
@vvysotskyi I've merge it to 1.18 and run in a testing environment, I didn't find any NullPointerException or other subsequent exceptions after merge. Drill can return data normally.
In my opinion, this class is mainly used to obtain the statistical information of each column in parquet. It may be necessary to keep the statistics of the columns. 
I will modify other questions about test cases according to your reviewed.

Reply:
It is used to collect metadata for partition columns only. Even if some specific field is not determined as a partition, it still may have its metadata and be used in row group pruning later. But if you will set the type to NULL, it may cause issues for this case when it is used later for some specific types like `FIXED_LEN_BYTE_ARRAY` or `BINARY`. So it is better do not mark such field as partition at all and avoid possible issues with determining comparator (see `ParquetGroupScanStatistics.getTypeForColumn()` method usage).

Reply:
@vvysotskyi 
I've add a if condition in `checkForPartitionColumn`. Do you mean like this?

Reply:
Not completely, in this case, no need to change this line, but add a simple check for `null` in `checkForPartitionColumn()` or before calling it (something like `majorType != null && checkForPartitionColumn...`).